File: app/src/main/java/com/jqorz/planewar/frame/GameSetting.java
package com.jqorz.planewar.frame;

import android.app.Activity;
import android.os.Bundle;
import android.widget.CompoundButton;
import android.widget.Switch;

import com.jqorz.planewar.R;
import com.jqorz.planewar.tools.UserDataManager;

public class GameSetting extends Activity implements CompoundButton.OnCheckedChangeListener {
    private Switch swt_Sound, swt_Music;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.game_setting);
        initView();
        initEvent();
        setSwitch();
    }


    private void setSwitch() {
        swt_Music.setChecked(UserDataManager.isOpenMusic());
        swt_Sound.setChecked(UserDataManager.isOpenSound());
    }

    private void initView() {
        swt_Sound = (Switch) findViewById(R.id.swt_Sound);
        swt_Music = (Switch) findViewById(R.id.swt_Music);

    }

    private void initEvent() {
        swt_Sound.setOnCheckedChangeListener(this);
        swt_Music.setOnCheckedChangeListener(this);
    }

    @Override
    public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
        switch (compoundButton.getId()) {
            case R.id.swt_Sound:
                UserDataManager.setOpenSound(b);
                break;
            case R.id.swt_Music:
                UserDataManager.setOpenMusic(b);
                break;
        }
    }

}

File: app/src/main/java/com/jqorz/planewar/frame/GameSetting.kt
package com.jqorz.planewar.frame

import android.app.Activity
import android.os.Bundle
import android.view.View
import android.widget.CompoundButton
import android.widget.Switch
import com.jqorz.planewar.R
import com.jqorz.planewar.tools.UserDataManager

class GameSetting : Activity(), CompoundButton.OnCheckedChangeListener {
    private var swt_Sound: Switch? = null
    private var swt_Music: Switch? = null
    override fun onCreate(savedInstanceState: Bundle) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.game_setting)
        initView()
        initEvent()
        setSwitch()
    }

    private fun setSwitch() {
        swt_Music!!.isChecked = UserDataManager.isOpenMusic()
        swt_Sound!!.isChecked = UserDataManager.isOpenSound()
    }

    private fun initView() {
        swt_Sound = findViewById<View>(R.id.swt_Sound) as Switch
        swt_Music = findViewById<View>(R.id.swt_Music) as Switch
    }

    private fun initEvent() {
        swt_Sound!!.setOnCheckedChangeListener(this)
        swt_Music!!.setOnCheckedChangeListener(this)
    }

    override fun onCheckedChanged(compoundButton: CompoundButton, b: Boolean) {
        when (compoundButton.id) {
            R.id.swt_Sound -> UserDataManager.setOpenSound(b)
            R.id.swt_Music -> UserDataManager.setOpenMusic(b)
        }
    }
}